apps/monarch/monarch-frontend/src/app/SearchTherapists.tsx

```diff
@@ -65,46 +65,249 @@ const debouncedSearchTherapists = debouncePromise(
+  const key = { phoneNumber: therapist.phone, fullName: therapist.fullName };
+
+  const [phone, setPhone] = useState<string>(therapist.phone);
+  const [fullName, setFullName] = useState<string>(therapist.fullName);
+  const [address, setAddress] = useState<string>(therapist.address);
+  const [email, setEmail] = useState<string>(therapist.email);
+  const [website, setWebsite] = useState<string | undefined>(therapist.website);
+  const [familiesHelped, setFamiliesHelped] = useState<number>(0);
+  const [dateJoined, setDateJoined] = useState<string>('2024-07-30');
+
+  const isFormValid = () => {
+    const dateFormat = /^\d{4}-\d{2}-\d{2}$/;
+    return (
+      phone.trim() !== '' &&
+      fullName.trim() !== '' &&
+      address.trim() !== '' &&
+      email.trim() !== '' &&
+      dateFormat.test(dateJoined)
+    );
+  };
+
+  const resetForm = () => {
+    setPhone(therapist.phone);
+    setFullName(therapist.fullName);
+    setAddress(therapist.address);
+    setEmail(therapist.email);
+    setWebsite(therapist.website);
+    setFamiliesHelped(0);
+  };
+
+  const handleClose = () => {
+    resetForm();
+    onEditClose();
+  };
+
+  const handleSave = () => {
+    if (isFormValid()) {
+      // TODO: save data in database
+      onEditClose();
+    }
+  };
+
+  return (
+    <>
+      <Button onClick={onEditOpen} colorScheme="teal" size="sm">
+        Edit
+      </Button>
+
+      <Modal isOpen={isEditOpen} onClose={onEditClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Edit Practitioner Info</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <VStack spacing={4}>
+              <FormControl isRequired>
+                <FormLabel>Phone</FormLabel>
+                <Input
+                  value={phone}
+                  onChange={(e) => setPhone(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Full Name</FormLabel>
+                <Input
+                  value={fullName}
+                  onChange={(e) => setFullName(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Address</FormLabel>
+                <Input
+                  value={address}
+                  onChange={(e) => setAddress(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Email</FormLabel>
+                <Input
+                  value={email}
+                  onChange={(e) => setEmail(e.target.value)}
+                />
+              </FormControl>
+              <FormControl isRequired>
+                <FormLabel>Date Joined (YYYY-MM-DD)</FormLabel>
+                <Input
+                  value={dateJoined}
+                  onChange={(e) => setDateJoined(e.target.value)}
+                />
+              </FormControl>
```

Comment:
Can we use a date picker here by setting `type='date'`? I didn't know this was a thing until now, it's pretty cool - it conveniently stores the date value in the format we want, and provides a nice UI that enforces the format as well. Here it is on firefox:

![image](https://github.com/user-attachments/assets/c7e6a278-e4c2-4d29-9e11-750ef0fc3520)